bukkit/src/main/java/com/github/games647/fastlogin/bukkit/listener/protocollib/VerifyResponseTask.java

```diff
@@ -171,17 +175,33 @@ private Object getNetworkManager() throws IllegalAccessException, ClassNotFoundE
     }
 
     private boolean enableEncryption(SecretKey loginKey) throws IllegalArgumentException {
+        if (encryptMethod == null) {
+            Class<?> networkManagerClass = MinecraftReflection.getNetworkManagerClass();
+
+            try {
+                encryptMethod = FuzzyReflection.fromClass(networkManagerClass)
+                        .getMethodByParameters("a", SecretKey.class);
+            } catch (IllegalArgumentException exception) {
+                encryptMethod = FuzzyReflection.fromClass(networkManagerClass)
+                        .getMethodByParameters("a", Cipher.class, Cipher.class);
+
+                Class<?> encryptionClass = MinecraftReflection.getMinecraftClass("MinecraftEncryption");
+                cipherMethod = FuzzyReflection.fromClass(encryptionClass)
+                        .getMethodByParameters("a", int.class, Key.class);
+            }
+        }
+
         try {
-            //get the NMS connection handle of this player
-            Object networkManager = getNetworkManager();
+            Object networkManager = this.getNetworkManager();
 
-            //try to detect the method by parameters
-            Method encryptMethod = FuzzyReflection
-                    .fromObject(networkManager).getMethodByParameters("a", SecretKey.class);
+            if (cipherMethod == null) {
+                encryptMethod.invoke(networkManager, loginKey);
+            } else {
+                Object cipher = cipherMethod.invoke(null, 2, loginKey);
```

Comment:
Please give them a more accurate name like how they are used.

Reply:
I assume you've meant the names and/or modes of Ciphers - I've changed both to make it all clearer

Reply:
I meant the variable name (something like `initializedCipher`) and/or a description of what the method does, but it's even better now.